appscale/tools/remote_helper.py

```diff
@@ -938,41 +940,45 @@ def terminate_virtualized_cluster(cls, keyname, is_verbose):
 
     acc = AppControllerClient(shadow_host, secret)
     try:
-      all_ips = acc.get_all_public_ips()
+      terminate_thread = threading.Thread(target=acc.run_terminate,
+                                          args=(clean,))
+      terminate_thread.start()
+      terminated_successfully = True
+      log_dump = ""
+      while not acc.is_appscale_terminated():
+        # for terminate receive_server_message will return a JSON string that
+        #  is a list of dicts with keys: ip, status, output
+        output_list = yaml.safe_load(acc.receive_server_message())
+        if output_list == "Error":
+          time.sleep(0.3)
```

Comment:
You know I think this wait time might not even be necessary anymore... I'll look into it.